core/app/models/spree/role.rb

```diff
@@ -3,12 +3,26 @@
 module Spree
   class Role < Spree::Base
     has_many :role_users, class_name: "Spree::RoleUser", dependent: :destroy
+    has_many :role_permissions, dependent: :destroy
+    has_many :permission_sets, through: :role_permissions
     has_many :users, through: :role_users
 
+    scope :non_base_roles, -> { where.not(name: ['admin']) }
+
     validates_uniqueness_of :name, case_sensitive: true
+    validates :name, uniqueness: true
+    after_save :assign_permissions
 
     def admin?
       name == "admin"
     end
+
+    def permission_sets_constantized
+      permission_sets.map(&:set).map(&:constantize)
+    end
+
+    def assign_permissions
+      ::Spree::Config.roles.assign_permissions name, permission_sets_constantized
```

Comment:
If I'm not missing something, that will only update the permissions in-memory, and it won't survive an application reload https://github.com/solidusio/solidus/blob/83a62431494d8a8ed0c241247c43d3dc37bba98d/core/lib/spree/core/role_configuration.rb#L67.

Reply:
You're right! 
This is an important thing that I've added on the [backend interface pull request](https://github.com/nebulab/solidus/pull/362/files#diff-8f107b502c1d91ce09dcb78e3514f1e4503a7276f98a82a5f545281843e3e34a) that I forgot (and must) bring to this one!